bookkeeper-http/http-server/src/main/java/org/apache/bookkeeper/http/HttpServiceProvider.java

```diff
@@ -0,0 +1,131 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.bookkeeper.http;
+
+import org.apache.bookkeeper.http.service.HttpService;
+
+/**
+ * Provider to provide services for different http endpoint handlers.
+ */
+public interface HttpServiceProvider {
+
+    /**
+     * Provide heartbeat service for heartbeat api.
+     */
+    HttpService provideHeartbeatService();
```

Comment:
same as comment above we could have a single method with a type parameter (an enum)

Reply:
Thanks.  commented there.